rotkehlchen/chain/evm/decoding/velodrome/velodrome_cache.py

```diff
@@ -204,15 +256,18 @@ def query_velodrome_data_from_chain_and_maybe_create_tokens(
     return returned_pools
 
 
-def query_velodrome_data(inquirer: 'OptimismInquirer') -> list[VelodromePoolData] | None:
+def query_velodrome_like_data(
+        inquirer: 'OptimismInquirer | BaseInquirer',
```

Comment:
wondering why not add CacheType as another argument and avoid the if? Similar questions for above? Not really suggesting it, but perhaps it would be beneficial?

Reply:
In `ReloadablePoolsAndGaugesDecoderMixin` we provide it as argument and then we would need to provide the cache type in kwargs. This argument seems to be used by all except convex. I'll provide it as an optional argument